proposals/1406_multiple-idps/README.md

```diff
@@ -0,0 +1,493 @@
+       # Any compilation or type-checking failure of any expression will cause an error status on the FederationDomain.
+       # Any unexpected runtime evaluation errors (e.g. division by zero) cause the login to fail.
+       # When all expressions evaluate successfully, then the username and groups has been decided for that login.
+       expressions:
+         # This expression runs first, so it operates on unmodified usernames and groups as extracted from the IDP.
+         # It rejects auth for any user who does not belong to certain groups.
+         - type: policy/v1
+           expression: 'groups.exists(g, g in strListConst.mustBelongToOneOfThese)'
+           message: "Only users in certain kube groups are allowed to authenticate"
+         # This expression runs second, and the previous expression was a policy (which cannot change username or
+         # groups), so this expression also operates on the unmodified usernames and groups as extracted from the
+         # IDP. For certain users, this adds a new group to the list of groups.
+         - type: groups/v1
+           expression: 'username in strListConst.additionalAdmins ? groups + ["kube/admins"] : groups'
+         # This expression runs next. Due to the expression above, this expression operates on the original username,
+         # and on a potentially changed list of groups. This drops all groups which do not start with a certain prefix.
+         - type: groups/v1
+           expression: 'groups.filter(group, group.startsWith(onlyIncludeGroupsWithThisPrefix))'
+         # Due to the expressions above, this expression operates on the original username, and on a potentially
+         # changed list of groups. This unconditionally prefixes the username.
+         - type: username/v1
+           expression: 'strConst.prefix + username'
+         # The expressions above have already changed the username and might have changed the groups before this
+         # expression runs. This unconditionally prefixes all group names.
+         - type: groups/v1
+           expression: 'groups.map(group, strConst.prefix + group)'
+
+       # Examples can optionally be used to ensure that the above sequence of expressions is working as expected.
+       # Examples define sample input identities which are then run through the above expression list,
+       # and the results are compared to the expected results. If any example in this list fails, then this
+       # FederationDomain will not be available for use, and the error(s) will be added to its status.
+       # This can be used to help guard against programming mistakes in the above CEL expressions, and also
+       # act as living documentation for other administrators to better understand the above CEL expressions.
+       examples:
+         - username: ryan@example.com
+           groups: [ kube/developers, kube/auditors, non-kube-group ]
+           expects:
+              username: ryan@example.com
```

Comment:
This is a typo. Should be `ad:ryan@example.com`, since there is a transformation that always prepends the prefix to usernames.

Reply:
Fixed and force pushed.